### ISSUE TITLE
NoClassDefFoundError after bumping to selenium-java 4.0.0 in Mac OS

### ISSUE THREAD

**Issue by bonigarcia** [SeleniumHQ/selenium#9784]
## 🐛 Bug Report

I have a couple of repositories using Selenium 4:

* https://github.com/bonigarcia/selenium-webdriver-java/
* https://github.com/bonigarcia/webdrivermanager/

I use GitHub Actions to run the tests in these repositories, using three different [virtual environments](https://github.com/actions/virtual-environments):

* `ubuntu-latest` (Ubuntu 20.04).
* `windows-latest` (Windows Server 2019).
* `macos-latest` (macOS 10.15).

After bumping to selenium-java 4.0.0-rc-1, several tests of the Mac OS workflow are failing due to `NoClassDefFoundError`, such as:

```
java.lang.NoClassDefFoundError: Could not initialize class org.openqa.selenium.WebDriverException
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:571)
```

... or:

```
java.lang.NoClassDefFoundError: Could not initialize class org.openqa.selenium.net.HostIdentifier
	at io.github.bonigarcia.wdm.test.create.ChromeRemoteTest.setupClass(ChromeRemoteTest.java:41)
```

## To Reproduce/Test script

I don't have a Mac OS machine to test it, but the error seems consistent in Mac OS (at least on GitHub Actions).

The test that fails are the following:

* 10 tests in selenium-webdriver-java. For instance: [FluentWaitJUnit4Test](https://github.com/bonigarcia/selenium-webdriver-java/blob/master/selenium-webdriver-junit4/src/test/java/io/github/bonigarcia/webdriver/junit4/ch3/wait/FluentWaitJUnit4Test.java) (see the complete [workflow traces](https://github.com/bonigarcia/selenium-webdriver-java/actions/runs/1193944103) and the [workflow setup](https://github.com/bonigarcia/selenium-webdriver-java/blob/master/.github/workflows/build.yml)).
* 1 test in WebDriverManager: [ChromeRemoteTest](https://github.com/bonigarcia/webdrivermanager/blob/master/src/test/java/io/github/bonigarcia/wdm/test/create/ChromeRemoteTest.java) (see [workflow traces](https://github.com/bonigarcia/webdrivermanager/actions/runs/1193898533) and the [workflow setup](https://github.com/bonigarcia/webdrivermanager/blob/master/.github/workflows/build.yml)).

## Environment

OS: macOS 10.15 (on GitHub Actions)
Browser: Chrome
Browser version: 93
Browser Driver version: chromedriver 93.0.4577.15
Language Bindings version: selenium-java 4.0.0-rc-1
Selenium Grid version (if applicable): 4.0.0-rc-1 (the dependency `org.seleniumhq.selenium:selenium-grid` is also used in WebDriverManager, but not in selenium-webdriver-java)

**Comment by bonigarcia** [comment]
I continued looking into this issue. It still seems persistent when using Mac OS in GitHub Actions. In the last executions of my repos, the lastest version of chromedriver 93 (i.e., 93.0.4577.63) is used, and the problem still happens:

https://github.com/bonigarcia/selenium-webdriver-java/runs/3574087304?check_suite_focus=true

The stack trace seems to point to [ElementLocation.java#L162](https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/remote/ElementLocation.java#L162), i.e., when exchanging some command with the driver:

```
java.lang.NoClassDefFoundError: Could not initialize class org.openqa.selenium.WebDriverException
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:571)
	at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:162)
	at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:66)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:381)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:373)
```

If I am not wrong, this should raise a `UnreachableBrowserException`, but for some reason I don't understand, `WebDriverException` is not found and it fails due to `NoClassDefFoundError`.

**Comment by bonigarcia** [comment]
The same issue happens with selenium-java 4.0.0-rc-2:

https://github.com/bonigarcia/selenium-webdriver-java/runs/3765276322?check_suite_focus=true

Does anybody know what could be the problem here? Thanks.

**Comment by bonigarcia** [comment]
I think I narrowed the problem by executing only two tests of my test suite. The steps to reproduce are the following (in macOS):

```
git clone https://github.com/bonigarcia/selenium-webdriver-java/
cd selenium-webdriver-java/selenium-webdriver-junit4
mvn test -Dtest=ImplicitWaitJUnit4Test,FluentWaitJUnit4Test
```

In the execution, most of the time (not always), the second test fails. It seems the browser crashes (or at least, it is closed) when trying to load the web page. The test, which theoretically should fail due to a `UnreachableBrowserException`, throws `NoClassDefFoundError` instead. The complete logging info is the following:

```
boni@bonis-iMac selenium-webdriver-junit4 % mvn test -Dtest=ImplicitWaitJUnit4Test,FluentWaitJUnit4Test
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------< io.github.bonigarcia:selenium-webdriver-junit4 >-----------
[INFO] Building selenium-webdriver-junit4 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ selenium-webdriver-junit4 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ selenium-webdriver-junit4 ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ selenium-webdriver-junit4 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/boni/Documents/dev/selenium-webdriver-java/selenium-webdriver-junit4/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ selenium-webdriver-junit4 ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ selenium-webdriver-junit4 ---
[INFO] Surefire report directory: /Users/boni/Documents/dev/selenium-webdriver-java/selenium-webdriver-junit4/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running io.github.bonigarcia.webdriver.junit4.ch3.wait.ImplicitWaitJUnit4Test
2021-10-06 22:18:46 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.checkKeyInResolutionCache(186) - Resolution chrome=94 in cache (valid until 14:18:11 06/10/2021 PDT)
2021-10-06 22:18:46 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.checkKeyInResolutionCache(186) - Resolution chrome94=94.0.4606.61 in cache (valid until 11:59:26 07/10/2021 PDT)
2021-10-06 22:18:46 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.resolveDriverVersion(966) - Using chromedriver 94.0.4606.61 (resolved driver for Chrome 94)
2021-10-06 22:18:46 [main] DEBUG i.g.bonigarcia.wdm.WebDriverManager.manage(922) - Driver chromedriver 94.0.4606.61 found in cache
2021-10-06 22:18:46 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.exportDriver(1021) - Exporting webdriver.chrome.driver as /Users/boni/.cache/selenium/chromedriver/mac64/94.0.4606.61/chromedriver
Starting ChromeDriver 94.0.4606.61 (418b78f5838ed0b1c69bb4e51ea0252171854915-refs/branch-heads/4606@{#1204}) on port 31969
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
Oct 06, 2021 10:18:57 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
Oct 06, 2021 10:18:57 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 94
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 20.656 sec
Running io.github.bonigarcia.webdriver.junit4.ch3.wait.FluentWaitJUnit4Test
2021-10-06 22:19:06 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.checkKeyInResolutionCache(186) - Resolution chrome=94 in cache (valid until 14:18:11 06/10/2021 PDT)
2021-10-06 22:19:06 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.checkKeyInResolutionCache(186) - Resolution chrome94=94.0.4606.61 in cache (valid until 11:59:26 07/10/2021 PDT)
2021-10-06 22:19:06 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.resolveDriverVersion(966) - Using chromedriver 94.0.4606.61 (resolved driver for Chrome 94)
2021-10-06 22:19:06 [main] DEBUG i.g.bonigarcia.wdm.WebDriverManager.manage(922) - Driver chromedriver 94.0.4606.61 found in cache
2021-10-06 22:19:06 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.exportDriver(1021) - Exporting webdriver.chrome.driver as /Users/boni/.cache/selenium/chromedriver/mac64/94.0.4606.61/chromedriver
Starting ChromeDriver 94.0.4606.61 (418b78f5838ed0b1c69bb4e51ea0252171854915-refs/branch-heads/4606@{#1204}) on port 6446
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
Oct 06, 2021 10:19:10 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
Oct 06, 2021 10:19:11 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Found exact CDP implementation for version 94
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 5.471 sec <<< FAILURE!
testFluentWait(io.github.bonigarcia.webdriver.junit4.ch3.wait.FluentWaitJUnit4Test)  Time elapsed: 5.471 sec  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.openqa.selenium.WebDriverException
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:583)
	at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:162)
	at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:66)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:383)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:375)
	at org.openqa.selenium.support.ui.ExpectedConditions$6.apply(ExpectedConditions.java:174)
	at org.openqa.selenium.support.ui.ExpectedConditions$6.apply(ExpectedConditions.java:171)
	at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:208)
	at io.github.bonigarcia.webdriver.junit4.ch3.wait.FluentWaitJUnit4Test.testFluentWait(FluentWaitJUnit4Test.java:59)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)


Results :

Tests in error: 
  testFluentWait(io.github.bonigarcia.webdriver.junit4.ch3.wait.FluentWaitJUnit4Test): Could not initialize class org.openqa.selenium.WebDriverException

Tests run: 2, Failures: 0, Errors: 1, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.166 s
[INFO] Finished at: 2021-10-06T22:19:12+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project selenium-webdriver-junit4: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/boni/Documents/dev/selenium-webdriver-java/selenium-webdriver-junit4/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
boni@bonis-iMac selenium-webdriver-junit4 % 
```

The problem happens with selenium-java 4.0.0-rc-1 and 4.0.0-rc-2. Nevertheless, if I downgrade to 4.0.0-beta-4 in the [pom.xml](https://github.com/bonigarcia/selenium-webdriver-java/blob/master/selenium-webdriver-junit4/pom.xml), the issue never happens.

**Comment by zhuyuemings** [comment]
Same Error Happens on MacOS

  ```
Exception in thread "main" java.lang.NoClassDefFoundError: org.openqa.selenium.WebDriverException
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:583)
        at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElements(ElementLocation.java:179)
        at org.openqa.selenium.remote.ElementLocation.findElements(ElementLocation.java:108)
        at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:398)
        at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:390)
```

It Happens when i use By.className() in Java and it send {"using":"class"} first, driver returns "invalid argument"

then when run to createException in W3CHttpResponseCodec it throws NoClassDefFoundError

```
    private WebDriverException createException(String error, String message) {
      Class<? extends WebDriverException> clazz = errorCodes.getExceptionType(error);
      try {
        Constructor<? extends WebDriverException> constructor = clazz.getConstructor(String.class);//happens on this line
        return constructor.newInstance(message); 
      } catch (ReflectiveOperationException e) {
        throw new WebDriverException(message);
      }
    }
```

finally run to execute catch cause in RemoteWebDriver

```
try {
      log(sessionId, command.getName(), command, When.BEFORE);
      response = executor.execute(command);
      log(sessionId, command.getName(), response, When.AFTER);

      if (response == null) {
        return null;
      }

      // Unwrap the response value by converting any JSON objects of the form
      // {"ELEMENT": id} to RemoteWebElements.
      Object value = getElementConverter().apply(response.getValue());
      response.setValue(value);
    } catch (Throwable e) {
      log(sessionId, command.getName(), command, When.EXCEPTION);
      WebDriverException toThrow;
      if (command.getName().equals(DriverCommand.NEW_SESSION)) {
        if (e instanceof SessionNotCreatedException) {
          toThrow = (WebDriverException) e;
        } else {
          toThrow = new SessionNotCreatedException(
            "Possible causes are invalid address of the remote server or browser start-up failure.", e);
        }
      } else if (e instanceof WebDriverException) {
        toThrow = (WebDriverException) e;
      } else {
        toThrow = new UnreachableBrowserException(
          "Error communicating with the remote browser. It may have died.", e); // finally happens here
      }
      populateWebDriverException(toThrow);
      toThrow.addInfo("Command", command.toString());
      throw toThrow;
    } finally {
      Thread.currentThread().setName(currentName);
    }
```

it seems like can not find UnreachableBrowserException and InvalidArgumentException 's parent class WebDriverException

i don't know why, i check the pom dependencies, didn't find anything wrong

**Comment by zhuyuemings** [comment]
I use

`Thread.currentThread().getContextClassLoader().findClass("org.openqa.selenium.WebDriverException")`

and it returns 

`
loader (instance of  sun/misc/Launcher$AppClassLoader): attempted  duplicate class definition for name: "org/openqa/selenium/WebDriverException"
`

**Comment by zhuyuemings** [comment]
I resolved this problem by add "HostIdentifier.getHostAddress();" at the beginning of my program

I think this problem is due to the static code in HostIdentifier

```
static {
    // Ideally, we'd use InetAddress.getLocalHost, but this does a reverse DNS lookup. On Windows
    // and Linux this is apparently pretty fast, so we don't get random hangs. On OS X it's
    // amazingly slow. That's less than ideal. Figure things out and cache.
    String host = System.getenv("HOSTNAME");  // Most OSs
    if (host == null) {
      host = System.getenv("COMPUTERNAME");  // Windows
    }
    if (host == null && Platform.getCurrent().is(Platform.MAC)) {
      try {
        Process process = Runtime.getRuntime().exec("hostname");

        if (!process.waitFor(2, TimeUnit.SECONDS)) {
          process.destroyForcibly();
          // According to the docs for `destroyForcibly` this is a good idea.
          process.waitFor(2, TimeUnit.SECONDS);
        }
        if (process.exitValue() == 0) {
          try (InputStreamReader isr = new InputStreamReader(process.getInputStream(), Charset.defaultCharset());
               BufferedReader reader = new BufferedReader(isr)) {
            host = reader.readLine();
          }
        }
      } catch (InterruptedException e) {
        Thread.currentThread().interrupt();
        throw new RuntimeException(e);
      } catch (Exception e) {
        // fall through
      }
    }
    if (host == null) {
      // Give up.
      try {
        host = InetAddress.getLocalHost().getHostName();
      } catch (Exception e) {
        host = "Unknown";  // At least we tried.
      }
    } 
```

but i don't know what happens ~

**Comment by bonigarcia** [comment]
This problem also happens with selenium-java 4.0.0:

https://github.com/bonigarcia/selenium-webdriver-java/runs/3884606418?check_suite_focus=true

**Comment by asolntsev** [comment]
@bonigarcia @zhuyuemings I managed to reproduce the issue once, here is my stacktrace:

```java
java.lang.NoClassDefFoundError: Could not initialize class org.openqa.selenium.net.HostIdentifier
	at integration.CollectionReloadingTest.reloadsCollectionOnEveryCall(CollectionReloadingTest.java:28)
```

Some error happens in the static initialisation block of class `HostIdentifier`. Unfortunately, I don't see the exact error, and cannot reproduce it anymore. But it's clear how to fix it: just add more try/catch blocks there.

P.S. As I already suggested in https://github.com/SeleniumHQ/selenium/issues/3048, the host resolving is not really needed and should be removed from Selenium code.

**Comment by asolntsev** [comment]
@shs96c @barancev I prepared a PR: https://github.com/SeleniumHQ/selenium/pull/9931

**Comment by asolntsev** [comment]
I got the original stack trace causing `NoClassDefFoundError`:

```java
java.lang.InterruptedException
	at java.base/java.lang.Object.wait(Native Method)
	at java.base/java.lang.Object.wait(Object.java:472)
	at java.base/java.util.concurrent.TimeUnit.timedWait(TimeUnit.java:408)
	at java.base/java.lang.ProcessImpl.waitFor(ProcessImpl.java:449)
	at org.openqa.selenium.net.HostIdentifier.<clinit>(HostIdentifier.java:46)
	at org.openqa.selenium.WebDriverException.<clinit>(WebDriverException.java:34)
	at org.openqa.selenium.os.CommandLine.waitFor(CommandLine.java:130)
	at org.openqa.selenium.remote.service.DriverService.lambda$start$2(DriverService.java:208)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by diemol** [comment]
Is something else needed after merging the PR?

**Comment by asolntsev** [comment]
@diemol Yes, now it would be great to release the fix. :)

**Comment by diemol** [comment]
If things go well, we will release next week.
You should join us in our Slack channel to stay even more up to date!

**Comment by asolntsev** [comment]
@bonigarcia @zhuyuemings I found even simpler way to reproduce the issue:

```java
  @Test
  void reproduceSeleniumIssue9784() {
    Thread.currentThread().interrupt();
    try {
      throw new WebDriverException("this causes ExceptionInInitializerError");
    }
    catch (ExceptionInInitializerError e) {
      throw new WebDriverException("this causes NoClassDefFoundError");
    }
  }
```

The output is:
```java
java.lang.NoClassDefFoundError: Could not initialize class org.openqa.selenium.WebDriverException
```

**Comment by asolntsev** [comment]
Let me clarify the status of this issue. Though its status is "Closed", it's not entirely solved in Selenium 4.1.0 yet. 

This issue has 2 related pull requests:
* PR #9931  - improved the error logging and replaced `NoClassDefFoundError` by a better exception, but didn’t solve the problem entirely. Released in Selenium 4.1.0.
* PR #10063  - does finally solve the problem. It’s was recently merged to trunk, but **not released yet**.

**Comment by bonigarcia** [comment]
I can confirm the new release 4.1.0 works better than the former 4.0.0 (several of my broken tests now pass with the new release), but now there are tons of the following traces (when running in macOS):

```
WARNING: Failed to resolve host name
java.lang.InterruptedException
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:460)
	at java.util.concurrent.TimeUnit.timedWait(TimeUnit.java:348)
	at java.lang.UNIXProcess.waitFor(UNIXProcess.java:410)
	at org.openqa.selenium.net.HostIdentifier.resolveHostName(HostIdentifier.java:52)
	at org.openqa.selenium.net.HostIdentifier.getHostName(HostIdentifier.java:116)
	at org.openqa.selenium.WebDriverException.getSystemInformation(WebDriverException.java:95)
	at org.openqa.selenium.WebDriverException.createMessage(WebDriverException.java:87)
	at org.openqa.selenium.WebDriverException.getMessage(WebDriverException.java:65)
	at java.lang.Throwable.getLocalizedMessage(Throwable.java:392)
	at java.lang.Throwable.toString(Throwable.java:481)
	at java.lang.Throwable.<init>(Throwable.java:312)
	at java.lang.Exception.<init>(Exception.java:102)
	at java.lang.RuntimeException.<init>(RuntimeException.java:96)
	at java.util.concurrent.CompletionException.<init>(CompletionException.java:88)
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1606)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Error: Exception in thread "Driver Service Executor" java.lang.RuntimeException: java.lang.InterruptedException
	at org.openqa.selenium.net.HostIdentifier.resolveHostName(HostIdentifier.java:67)
	at org.openqa.selenium.net.HostIdentifier.getHostName(HostIdentifier.java:116)
	at org.openqa.selenium.WebDriverException.getSystemInformation(WebDriverException.java:95)
	at org.openqa.selenium.WebDriverException.createMessage(WebDriverException.java:87)
	at org.openqa.selenium.WebDriverException.getMessage(WebDriverException.java:65)
	at java.lang.Throwable.getLocalizedMessage(Throwable.java:392)
	at java.lang.Throwable.toString(Throwable.java:481)
	at java.lang.Throwable.<init>(Throwable.java:312)
	at java.lang.Exception.<init>(Exception.java:102)
	at java.lang.RuntimeException.<init>(RuntimeException.java:96)
	at java.util.concurrent.CompletionException.<init>(CompletionException.java:88)
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1606)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:460)
	at java.util.concurrent.TimeUnit.timedWait(TimeUnit.java:348)
	at java.lang.UNIXProcess.waitFor(UNIXProcess.java:410)
	at org.openqa.selenium.net.HostIdentifier.resolveHostName(HostIdentifier.java:52)
	... 16 more
Nov 24, 2021 5:50:22 PM org.openqa.selenium.net.HostIdentifier resolveHostName
```

**Comment by asolntsev** [comment]
@bonigarcia Exactly. It is the error that was swallowed in Selenium 4.0.0 and caused `NoClassDefFoundError`. Now it's at least visible in Selenium 4.1.0. Will be fixed by PR #10063.

P.S. You are lucky! I tried to reproduce this problem many times, but didn't succeed. :)
